File: src/geometryService/cpp/stractNet/SNRoot.cpp
///////////////////////////////////////////////////////////
//  SNRoot.cpp
//  Implementation of the Class SNRoot
//  Created on:      06-Aug-2008 7:50:22 AM
//  Original author: david.h.loman
///////////////////////////////////////////////////////////

#include "SNRoot.h"

///////////////////////////////////////////////////////////////////////
/*
 *
 * xSTRUCTORS
 *
 */
///////////////////////////////////////////////////////////////////////

SNRoot::SNRoot(String Name) {
	this->ObjName = Name;
	this->MS_local = SNRoot::MS_Global;
}

SNRoot::~SNRoot() {

}

///////////////////////////////////////////////////////////////////////
/*
 *
 * METHODS
 *
 */
///////////////////////////////////////////////////////////////////////

MessagingSystem& SNRoot::getLocalMS() {
	return this->MS_Local;
}

void SNRoot::setLocalMS(MessagingSystem& ms) {
	this->MS_Local = ms;
}

/**
 * Quick & Dirty getter'n'setters -> simplifies inclusion into Strings
 */
String SNRoot::gON() {
	return this->ObjName;
}
void SNRoot::sON(String name) {
	this->ObjName = name;
}

//TODO clean up the internal STDOUT and STDERR
void SNRoot::err(String message) {
	SNRoot::serr(this->gON() + ":\t" + message + "\n");
}

void SNRoot::out(String message) {
	SNRoot::sout(this->gON() + ":\t" + message + "\n");
}

///////////////////////////////////////////////////////////////////////
/*
 *
 * STATICS
 *
 */
///////////////////////////////////////////////////////////////////////


//TODO Eliminate the local and global MS passing via statics. Possible setup Singleton config
void SNRoot::setGlobalMS(MessagingSystem& ms) {
	SNRoot::MS_Global = ms;
}

void SNRoot::serr(String message) {
	cerr << message;
}

void SNRoot::sout(String message) {
	cout << message;
}

MessagingSystem& SNRoot::getGlobalMS() {
	return SNRoot::MS_Global;
}

File: src/iBME/superceded_GS/java/stractNet/src/stractNet/SNRoot.java
package stractNet;

import stractNet.messaging.MessagingSystem;

public abstract class SNRoot {
	public SNRoot (String Name) {
		this.ObjName = Name;
		this.MS_Local = SNRoot.MS_Global;
	}
	
	
	/*
	 * 0 = Normal Waypoints
	 * 1 = Object Status reports (NON-Debuging)
	 * 2 = Object Data output (Debuging)
	 * 
	 */
	public static int VerbosityLevel = 0;
	
	protected String ObjName = "";

	
	
	
	
	//Messaging System configurations
	private static MessagingSystem MS_Global;
	private MessagingSystem MS_Local = SNRoot.MS_Global;
	
	public static MessagingSystem getGlobalMS() {		
		return SNRoot.MS_Global;
	}

	public static void setGlobalMS(MessagingSystem ms) {
		SNRoot.MS_Global = ms;
	}

	
	public MessagingSystem getLocalMS() {		
		return this.MS_Local;
	}

	public void setLocalMS(MessagingSystem ms) {
		this.MS_Local = ms;
	}

	
	
	
	
	
	protected void out(String message, int vLevel) {
		SNRoot.sout(gON() + ":\t" + message, vLevel);
	}

	protected void err(String message, int vLevel) {
		SNRoot.serr(gON() + ":\t" + message, vLevel);
	}

	// TODO Clean up out,err,sout,err routing.
	public static void sout(String message, int vLevel) {
		if (vLevel <= SNRoot.VerbosityLevel) {
			synchronized(System.out) {
				System.out.println(message);
			}
		}
	}

	public static void serr(String message, int vLevel) {
		if (vLevel <= SNRoot.VerbosityLevel) {
			synchronized(System.err) {
				System.err.println(message);
			}
		}
	}

	
	
	
	
	
	// Quick & Dirty getter'n'setters -> simplifies inclusion into Strings
	public String gON() {
		return this.ObjName;
	}
	public void sON(String name) {
		this.ObjName = name;
	}
	
	public String gOT() {
		return this.getClass().getSimpleName();
	}
}
